### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by bsmiles32**

### CODE REVIEW

File: src/sdl_backend.c [hunk 1]
```diff
@@ -346,35 +364,36 @@ void sdl_push_samples(struct sdl_backend* sdl_backend, const void* src, size_t s
...
+    size = (size / 4) * 4;
 
+    /* We need to lock audio before accessing cbuff */
+    SDL_LockAudio();
+    unsigned char* dst = cbuff_head(&sdl_backend->primary_buffer, &available);
     if (size <= available)
     {
-        SDL_LockAudio();
-
         if (sdl_backend->swap_channels) {
```
Comment (richard42) [inline src/sdl_backend.c:378]: shouldn't this be:
if (!sdl_backend->swap_channels)

**Review by richard42 — CHANGES_REQUESTED** [review]

File: src/sdl_backend.c (hunk 1 above)
Comment (bsmiles32) [reply src/sdl_backend.c:378]: I didn't touch that logic here, but you're right the logic is only good for LittleEndian host, for BigEndian host the logic should be !swap_channels.
I can do a separate PR for that.
Explanation:
- N64 produces 16bit L/R samples in RDRAM that we can name "Lh Ll Rh Rl" (Left high/low, Right high/low).
- mupen organize RDRAM such that a 32bit word can be read by the host in a single instruction (no swapping). So for BigEndian host, bytes are ordered the same and for LittleEndian host bytes are "Rl Rh Ll Lh".
- memcpy doesn't reorder the bytes, so for BigEndian host we end up with "Lh Ll Rh Rl" which is non-swapped, but "Rl Rh Ll Lh" for LittleEndian host which is swapped
- The other branch swaps consecutive 16-bit hwords, so for BigEndian host L/R get swapped, and for LittleEndian host we end up with "Ll Lh Rl Rh" which is non swapped L/R samples.
So yeah, logic is wrong for BigEndian host, but OK for LittleEndian host.

Comment (bsmiles32) [reply src/sdl_backend.c:378]: See PR #37 for a possible fix

**Comment by Jj0YzL5nvJ** [comment]
Current master is broken with the current core without this patch...

**Pull request merged**